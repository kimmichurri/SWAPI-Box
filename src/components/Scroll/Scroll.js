import React from 'react';

export default function Scroll({ openingFilm }) {
    return (
        <section className="scroll-container">
            <div className="scroll">
                <h3 className="scroll-header">{ openingFilm.title }</h3>
                <p className="scroll-text"> { openingFilm.date }</p>
                <p className="scroll-text"> { openingFilm.crawl }</p>
            </div>
        </section>
    )
}