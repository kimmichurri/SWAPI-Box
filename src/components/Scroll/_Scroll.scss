.scroll-header {
    font-size: 80px;
}

.scroll-text {
    font-size: 60px;
}

.scroll-container {
    color: $white;
    display: flex;
    font-family: $container-header-font;
    height: 800px;
    justify-content: center;
    perspective: 200px;
}

.scroll {
    position: relative;
    transform-origin: 50% 100%;
    animation: scroll 90s linear;
}

@keyframes scroll {
    0% {
        top: 0;
        transform: rotateX(20deg) translateZ(0);
    }
    100% {
        top: -6000px;
        transform: rotateX(25deg) translateZ(-2500px);
    }
}