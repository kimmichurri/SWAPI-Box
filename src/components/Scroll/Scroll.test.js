import React from 'react';
import { shallow } from 'enzyme';
import Scroll from './Scroll'

describe('Scroll', () => {
  let wrapper;
  const mockOpeningFilm = {
    title: 'Movie',
    date: 2019,
    crawl: 'crawl'
  }

  beforeEach(() => {
    wrapper = shallow(<Scroll openingFilm={mockOpeningFilm} />)
  })

  it('should match the snapshot', () => {
    expect(wrapper.debug()).toMatchSnapshot()
  })
})