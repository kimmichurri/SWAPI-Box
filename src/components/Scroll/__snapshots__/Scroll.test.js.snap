// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scroll should match the snapshot 1`] = `
"<section className=\\"scroll-container\\">
  <div className=\\"scroll\\">
    <h3 className=\\"scroll-header\\">
      Movie
    </h3>
    <p className=\\"scroll-text\\">
       
      2019
    </p>
    <p className=\\"scroll-text\\">
       
      crawl
    </p>
  </div>
</section>"
`;
